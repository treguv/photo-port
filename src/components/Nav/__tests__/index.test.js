import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "../../Nav";

afterEach(cleanup);

describe("Nav component", () => {
  //baseline test
  it("renders", () => {
    render(<Nav />);
  });
  //snapshot test
  it("matches snapshot", () => {
    const { asFragment } = render(<Nav />);

    expect(asFragment).toMatchSnapshot();
  });
});

//test if emote is visible
describe("emoji is visible", () => {
  it("inserts emoji into h2", () => {
    //arrange
    const { getByLabelText } = render(<Nav />);
    //assert
    expect(getByLabelText("camera")).toHaveTextContent("ðŸ“¸");
  });
});

describe("links are visible", () => {
  it("inserts text into the links", () => {
    // Arrange
    const { getByTestId } = render(<Nav />);
    // Assert
    expect(getByTestId("link")).toHaveTextContent("Oh Snap!");
    expect(getByTestId("about")).toHaveTextContent("About me");
  });
});
