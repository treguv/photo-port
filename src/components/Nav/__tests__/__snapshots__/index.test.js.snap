// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `[Function]`;
